### PR TITLE
New function emscripten_has_asyncify

### PR DESCRIPTION
**Pull request by Beuc**
Cf. https://github.com/emscripten-ports/SDL2/issues/70#issuecomment-528447701

### CODE REVIEW

**Comment by gabrielcuvillier** [comment]
Good idea. 

Actually, I can think of a way to already get the info:

emscripten_get_compiler_setting("ASYNCIFY")

It needs -s RETAIN_COMPILER_SETTINGS=1 though

**Comment by sbc100** [comment]
True, but I wonder what the space cost of RETAIN_COMPILER_SETTINGS is?

**Comment by Beuc** [comment]
Also `emscripten_get_compiler_setting` hits a JavaScript error if not compiled-in.
Since we want to use the new function in the SDL2 port by default, that's annoying, and requires an update to existing build systems that use SDL2 outside of the port system (e.g. upstream SDL2 tarballs) so they add -s RETAIN_COMPILER_SETTINGS at link time. Not to mention people might not want to disclose their compiler settings :)

To a lesser extent the new functions also currently encapsulates ASYNCIFY||EMTERPRETER_ASYNC.

File: site/source/docs/api_reference/emscripten.h.rst [hunk 1]
```diff
@@ -992,6 +992,13 @@ Functions
   :returns: The value of the specified setting. Note that for values other than an integer, a string is returned (cast the ``int`` return value to a ``char*``).
   :rtype: int
 
+.. c:function:: int emscripten_has_asyncify()
+
+  Returns whether pseudo-synchronous functions can be used.
+
+  :rtype: int
+  :returns: 1 is program was compiled with ASYNCIFY=1 or EMTERPRETER_ASYNC=1, 0 otherwise.
```
Comment (kripken) [inline site/source/docs/api_reference/emscripten.h.rst]: `is` => `if`

File: src/library.js [hunk 2]
```diff
@@ -4258,6 +4258,14 @@ LibraryManager.library = {
     return cache[fullname] = allocate(intArrayFromString(ret + ''), 'i8', ALLOC_NORMAL);
   },
 
+  emscripten_has_asyncify: function() {
+#ifdef ASYNCIFY || EMTERPRETIFY_ASYNC
```
Comment (kripken) [inline src/library.js]: this can be `return {{{ ASYNCIFY || EMTERPRETIFY_ASYNC }}};` on one line

**Comment by Beuc** [comment]
Done!

**Comment by kripken** [comment]
Oh wait, tests are failing, that's odd.

Do you see the failures locally as well?

**Comment by Beuc** [comment]
`test_emscripten_has_asyncify` works locally for me.
Not sure what's happening :/

**Comment by Beuc** [comment]
I can reproduce with fastcomp which complains about ASYNCIFY being removed, while we need to set it for the testcase - I disabled the test in fastcomp.

**Comment by kripken** [comment]
Makes sense to disable it for fastcomp, yeah.

**Pull request merged**